File: Recipes/Recipes/Data/Entities/Recipe.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Recipes.Data.Entities
{
    public class Recipe
    {
        [Key]
        public int RecipeId { get; set; }
        public string RecipeName { get; set; }
        public string RecipeImage { get; set; }
        public int Portions { get; set; }
        public int Rating { get; set; }
        public string PrepTime { get; set; }
        public string CookTime { get; set; }
        public int? Calories { get; set; }
        public int? Fat { get; set; }
        public int? Saturated { get; set; }
        public int? Carbohydrates { get; set; }
        public int? Sugars { get; set; }
        public int? Fibre { get; set; }
        public int? Protein { get; set; }
        public int? Salt { get; set; }
        public ICollection<RecipeIngredient> Ingredients { get; set; }
        public ICollection<RecipeMethod> Methods { get; set; }
        public ICollection<RecipeNote> Notes { get; set; }
        public ICollection<RecipeCuisine> Cuisine { get; set; }
        public ICollection<RecipeTags> Tags { get; set; }
        public Skill Skill { get; set; }
        public Course Course { get; set; }
        public Category Category { get; set; }
    }
}

File: Recipes/Recipes/ClientApp/app/shared/recipe.ts
export class Recipe {
    recipeId: number;
    recipeName: string;
    recipeImage: string;
    portions: number;
    rating: number;
    prepTime: string;
    cookTime: string;
    calories: number;
    fat: number;
    saturated: number;
    carbohydrates: number;
    sugars: number;
    fibre: number;
    protein: number;
    salt: number;
    ingredients: Array<RecipeIngredient> = new Array<RecipeIngredient>();
    methods: Array<RecipeMethod> = new Array <RecipeMethod>();
    notes: Array<RecipeNote> = new Array <RecipeNote>();
    cuisine: Array<RecipeCuisine> = new Array<RecipeCuisine>();
    tags: Array<RecipeTag> = new Array<RecipeTag>();
    skillId: number;
    skillSkillLevel: string;
    courseId: number;
    courseName: string;
    categoryId: number;
    categoryName: string;
}

export class RecipeIngredient {
    id: number;
    quantity: number;
    ingredientId: number;
    ingredientName: string;
    measurementId: number;
    measurementMeasurement: string;
    preparationId: number;
    preparationPreparation: string;
}

export class RecipeMethod {
    id: number;
    stepNumber: number;
    method: string;
}

export class RecipeNote {
    id: number;
    note: string;
}

export class RecipeCuisine {
    id: number;
    cuisineId: number;
    cuisineName: string;
}

export class RecipeTag {
    id: number;
    tagId: number;
    tagName: string;
}